src/marker.rs

```diff
@@ -680,6 +696,55 @@ impl MarkerExpression {
             (MarkerValue::QuotedString(l_string), MarkerOperator::Equal, MarkerValue::Extra) => {
                 extras.contains(l_string)
             }
+            (
+                MarkerValue::MarkerEnvVersion(MarkerValueVersion::PythonVersion),
+                operator,
+                MarkerValue::QuotedString(r_string),
+            ) => {
+                // ignore all errors block
+                (|| {
+                    // The right hand side is allowed to contain a star, e.g. `python_version == '3.*'`
+                    let (r_version, r_star) = Version::from_str_star(r_string).ok()?;
+                    let operator = operator.to_pep440_operator()?;
+                    // operator and right hand side make the specifier
+                    let specifier = VersionSpecifier::new(operator, r_version, r_star).ok()?;
+
+                    // We're passing a lot of versions here, e.g. 3.8 .. 3.100, but since we're
+                    // passing them in order it is expected that we either match early or not at all
+                    let compatible = python_versions
+                        .iter()
+                        .any(|l_version| specifier.contains(l_version));
+                    Some(compatible)
+                })()
+                .unwrap_or(true)
+            }
+            (
+                MarkerValue::QuotedString(l_string),
+                operator,
+                MarkerValue::MarkerEnvVersion(MarkerValueVersion::PythonVersion),
+            ) => {
+                // ignore all errors block
+                (|| {
+                    // Not star allowed here, `'3.*' == python_version` is not a valid PEP 440
+                    // comparison
+                    let l_version = Version::from_str(l_string).ok()?;
+                    let operator = operator.to_pep440_operator()?;
+
+                    // We're passing a lot of versions here, e.g. 3.8 .. 3.100, but since we're
+                    // passing them in order it is expected that we either match early or not at all
+                    let compatible = python_versions.iter().any(|r_version| {
+                        // operator and right hand side make the specifier and in this case the
+                        // right hand is `python_version` so changes every iteration
+                        match VersionSpecifier::new(operator.clone(), r_version.clone(), false) {
+                            Ok(specifier) => specifier.contains(&l_version),
+                            Err(_) => true,
+                        }
+                    });
```

Comment:
I don't know if `VersionSpecifier` allows it but you could avoid cloning `operator` by using

```rust
for r_version in python_versions.iter().cloned() {
    match VersionSpecifier::new(operator, r_version, false) {
        Ok(specifier) => {
            if specifier.contains(&l_version) {
                return Some(true);
            } else {
                operator = specifier.operator
            }
        },
        Err(_) => {
            return Some(true);
        }
    }
}

```